src/Symfony/Component/Translation/Translator.php

```diff
@@ -311,6 +313,8 @@ public function getMessages($locale = null)
      */
     protected function loadCatalogue($locale)
     {
+        trigger_error('The '.__METHOD__.' method is deprecated since version 2.8 and will be removed in 3.0. Rely on the TranslatorBagInterface instead', E_USER_DEPRECATED);
```

Comment:
I suggest keeping this PR focused on ``getMessages`` only and deal with ``loadCatalogue`` later once we have a replacement for the extension point